dds/DCPS/transport/shmem/ShmemTransport.cpp

```diff
@@ -247,22 +250,34 @@ ShmemTransport::ReadTask::ReadTask(ShmemTransport* outer, ACE_sema_t semaphore)
 int
 ShmemTransport::ReadTask::svc()
 {
-  while (true) {
+  while (!stopped_) {
     ACE_OS::sema_wait(&semaphore_);
     if (stopped_) {
       return 0;
     }
     outer_->read_from_links();
   }
-  return 1;
+  return 0;
 }
 
 void
 ShmemTransport::ReadTask::stop()
 {
+  if (stopped_) {
+    return;
+  }
   stopped_ = true;
   ACE_OS::sema_post(&semaphore_);
-  wait();
+  ACE_Task_Base::wait();
```

Comment:
Why does it have the `ACE_Task_Base::` scope?

Reply:
There was a period where it didn't seem like wait was getting called so I made it explicit in case I was missing something. I'll change it back.